### DISCUSSION TITLE
[Preview 5] ExpressionObserver class is no longer accessible

### DISCUSSION THREAD

**Discussion by nsrahmad** [AvaloniaUI/Avalonia#10239] [Q&A] [answered]
**Describe the bug**
If using CommunityToolkit MVVM template and preview avalonia, the following line is present in `App.xaml.cs` to address #8397:
```csharp
ExpressionObserver.DataValidators.RemoveAll(x => x is DataAnnotationsValidationPlugin);
```
However, In preview 5, `ExpressionObserver` class is made internal. So I had to remove the above line. This makes the original issue come back. In my VM class, I have:
```csharp
public partial class MainWindowViewModel : ObservableValidator
{
 //...

    private DateTime? flightDate;

    [Required(ErrorMessage = "You must provide a valid flight date.")]
    public DateTime? FlightDate
    {
        get => flightDate;
        set
        {
            _ = SetProperty(ref flightDate, value, true);
            ValidateProperty(ReturnFlightDate, nameof(ReturnFlightDate));
        }
    }
 //...
```
and I see the Validation error twice:
![image](https://user-images.githubusercontent.com/10996250/216781268-4fae098d-d0d7-4669-bda9-6dd936d41ae6.png)
What is the proper way to fix this?
**To Reproduce**
The test project is at https://github.com/nsrahmad/7GUI-Avalonia/tree/master/FlightBooker

**Desktop (please complete the following information):**

- OS: Windows 10
- Version 11.0.0 Preview 5

**Comment by maxkatz6** [comment, marked as answer]
It's in `BindingPlugins.DataValidators` now

**Reply by X9VoiD** [reply]
It is, however `DataAnnotationsValidationPlugin` is also internal. Is there some new way to accomplish removing it?

**Reply by nsrahmad** [reply]
@X9VoiD Try:
```csharp
using Avalonia.Data.Core.Plugins;
...
BindingPlugins.DataValidators.RemoveAt(0);
```
```

**Reply by X9VoiD** [reply]
That feels too prone to breakage. What if for some reason the data validator at that index is not `DataAnnotationsValidationPlugin` anymore?

**Reply by X9VoiD** [reply]
I may use reflection for this but I'd rather not.

**Reply by nsrahmad** [reply]
Updated dotnet templates for Avalonia 11 do the same thing:
https://github.com/AvaloniaUI/avalonia-dotnet-templates/blob/5f07535d8eb4c34951f0b23fa1afd4ab1057949c/templates/csharp/app-mvvm/App.axaml.cs#L30

**Reply by Measurity** [reply]
@X9VoiD 

For similar effect you can do this:
```cs
for (int i = BindingPlugins.DataValidators.Count - 1; i >= 0; i--)
{
    if (BindingPlugins.DataValidators[i] is DataAnnotationsValidationPlugin)
    {
        BindingPlugins.DataValidators.RemoveAt(i);
    }
}
```
